modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/test/scala/edu/gemini/seqexec/web/server/security/FreeLDAPAuthenticationServiceSpec.scala

```diff
@@ -0,0 +1,65 @@
+package edu.gemini.seqexec.web.server.security
+
+import edu.gemini.seqexec.web.common.UserDetails
+import edu.gemini.seqexec.web.common.UserDetails._
+import org.scalatest.prop.PropertyChecks
+import org.scalatest.{FlatSpec, Matchers}
+
+import scalaz._
+import Scalaz._
+
+class FreeLDAPAuthenticationServiceSpec extends FlatSpec with Matchers with PropertyChecks {
+  import FreeLDAPAuthenticationService._
+
+  // Silly mock of a user database
+  case class MockAuthDB(users: Map[UID, (String, DisplayName)], acceptEmptyPwd: Boolean) {
+    def authenticate(u: String, p: String): UID = {
+      // This checks if the username and password but lets it bypass it
+      if (users.contains(u) && ((u == p && p.nonEmpty) || acceptEmptyPwd)) u else throw new RuntimeException()
+    }
+    def displayName(uid: UID): DisplayName = ~users.get(uid).map(_._2)
+  }
+
+  // Natural transformation to Id with a mock auth db
+  def toMockDB(db: MockAuthDB): LdapOp ~> Id =
```

Comment:
@swalker2m  In the case of an in memory database you don't need to know about the connection